### PR TITLE
HIVE-23244 Extract Create View analyzer from SemanticAnalyzer (Miklos Gergely)

### PR DESCRIPTION
**Pull request by miklosgergely**
Create View commands are not queries, but commands which have queries as a part of them. Therefore a separate CreateViewAnalyzer is needed which uses SemanticAnalyer to analyze it's query.

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  protected void validateReplaceWithPartitions(String viewName, Table oldView, List<FieldSchema> partitionColumns)
+      throws SemanticException {
+    String partitionViewErrorMsg = "The following view has partition, it could not be replaced: " + viewName;
+    try {
+      if (!oldView.getPartCols().isEmpty() && !db.getPartitions(oldView).isEmpty() &&
+          !oldView.getPartCols().equals(partitionColumns)) {
+        throw new SemanticException(ErrorMsg.REPLACE_VIEW_WITH_PARTITION.getMsg(partitionViewErrorMsg));
+      }
+    } catch (HiveException e) {
+      throw new SemanticException(ErrorMsg.REPLACE_VIEW_WITH_PARTITION.getMsg(partitionViewErrorMsg));
```
Comment (belugabehr) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java:117]: Again. this `SemanticException` will be double-wrapped

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AlterViewAsAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  private void validateCreateView(AlterViewAsDesc desc, SemanticAnalyzer analyzer) throws SemanticException {
+    try {
+      validateTablesUsed(analyzer);
+
+      //replace view
+      Table oldView = getTable(desc.getViewName(), false);
+      if (oldView == null) {
+        String viewNotExistErrorMsg = "The following view does not exist: " + desc.getViewName();
+        throw new SemanticException(
+            ErrorMsg.ALTER_VIEW_AS_SELECT_NOT_EXIST.getMsg(viewNotExistErrorMsg));
```
Comment (belugabehr) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AlterViewAsAnalyzer.java:84]: Catching your own exception

Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AlterViewAsAnalyzer.java]: I agree, fixed it.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java (hunk 1 above)
Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java]: I agree, fixed it.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        Table table = analyzer.getTableObjectByName(analyzer.getQB().getTabNameForAlias(alias));
+        if (table.isTemporary()) {
+          throw new SemanticException("View definition references temporary table " + alias);
+        }
+        if (table.isMaterializedView()) {
+          throw new SemanticException("View definition references materialized view " + alias);
+        }
+      } catch (HiveException ex) {
+        throw new SemanticException(ex);
+      }
```
[the remark this answers is not in the document]
Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java]: I agree, fixed it.

**Pull request merged**